docs/how_to_guide/20_time_series_embedding.ipynb

```diff
@@ -0,0 +1,205 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "021377e2",
+   "metadata": {},
+   "source": [
+    "## Using CausalCNNEmbedding"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9539c7dd",
+   "metadata": {},
+   "source": [
+    "We use `CausalCNNEmbedding` to extract **local temporal patterns** from the sequence and train a Neural Posterior Estimator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "981016ca",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define a causal CNN embedding for 1D time-series data\n",
+    "embedding_cnn = embedding_nets.CausalCNNEmbedding(\n",
+    "    input_shape=(100,),\n",
+    "    num_conv_layers=3,\n",
+    "    pool_kernel_size=10,\n",
+    "    output_dim=16,\n",
+    ")\n",
+    "\n",
+    "# Build density estimator with embedding\n",
+    "density_estimator_cnn = posterior_nn(\n",
+    "    model=\"maf\",\n",
+    "    embedding_net=embedding_cnn,\n",
+    "    z_score_x=\"none\",\n",
+    "    z_score_y=\"none\",\n",
+    ")\n",
+    "\n",
+    "# Create and train NPE using the prior and simulated data\n",
+    "inference_cnn = NPE(prior=prior, density_estimator=density_estimator_cnn)\n",
+    "posterior_cnn = inference_cnn.append_simulations(thetas, xs).train()\n",
+    "\n",
+    "# Draw posterior samples given an observed time series\n",
+    "samples_cnn = posterior_cnn.sample(torch.Size([10]), x_o)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "24905e86",
+   "metadata": {},
+   "source": [
+    "## Using TransformerEmbedding"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5a6aca03",
+   "metadata": {},
+   "source": [
+    "Next, we define a `TransformerEmbedding` that models **global dependencies** via self-attention."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a030cab7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Transformer configuration for sequence embedding\n",
+    "cfg = dict(\n",
+    "    vit=False,\n",
+    "    feature_space_dim=192,\n",
+    "    sequence_length=100,\n",
+    "    output_dim=16,\n",
+    "    num_layers=3,\n",
+    "    num_heads=12,\n",
+    "    head_dim=16,\n",
+    "    d_model=192,\n",
+    ")\n",
+    "\n",
+    "# Initialize the transformer embedding network\n",
+    "base_trans = embedding_nets.TransformerEmbedding(cfg)\n",
+    "\n",
+    "# Project 1D inputs to match transformer feature dimension\n",
+    "class ProjectedTransformer(nn.Module):\n",
+    "    def __init__(self, transformer):\n",
+    "        super().__init__()\n",
+    "        self.proj, self.transformer = nn.Linear(1, 192), transformer\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        if x.ndim == 2: \n",
+    "            x = x.unsqueeze(-1)\n",
+    "        x = self.proj(x)\n",
+    "        return self.transformer(x)\n",
+    "    \n",
+    "embedding_trans = ProjectedTransformer(base_trans)\n",
```

Comment:
Yeah, I noticed as well that the Transformer does not support 1D time series, and that it requires setting up the config externally. This is not user-friendly and we should fix it. Would you be up for creating an issue describing this problem and suggestion to support scalar time series as well and move the `config` into the init as explicit kwargs? That'd be great!